File: Module_01_Big_O/Topic_03_Linear_Time_O(n)/Python/example_02.py
# -*- coding: utf-8 -*-
"""
Course title: AlgoMania
Instructor: Thiago Brito - Universo DEV
Example adapted by: Charles Fernandes de Souza
Date: September 27, 2024
"""
def find_max(my_array):
  max = 0  
  for item in my_array:  # O(n) linear time, iterates over each element in the array
    if item > max:  
      max = item  
  return max  
  
def main():
  my_array = [10,20,30,40]  
  result = find_max(my_array)  # O(n) linear time due to the find_max function
  print(result)  
main() 

File: Module_01_Big_O/Topic_03_Linear_Time_O(n)/Java/Example02/src/com/souza/charles/Example02.java
package com.souza.charles;
/**
 Course title: AlgoMania
 Instructor: Thiago Brito - Universo DEV
 Example adapted by: Charles Fernandes de Souza
 Date: September 28, 2024
 */
public class Example02 {

    public static void main(String[] args) {

        int[] myArray = {10, 20, 30, 40};
        int result = findMax(myArray); // O(n) linear time due to the findMax function
        System.out.println(result);
    }

    private static int findMax(int[] myArray) {
        int max = 0;
        for (int item: myArray) { // O(n) linear time, iterates over each element in the array
            if (item > max) {
                max = item;
            }
        }
        return max;
    }
}